### PR TITLE
Simulation, clearing sphere, and ESDF fixes

### PR DESCRIPTION
**Pull request by helenol**
- Simulation world no longer crashes in very, very rare cases.
 - Fixed sphere computation to never miss voxels when centered near boundaries.
 - Fix issue https://github.com/ethz-asl/voxblox/issues/224 which was failing to mark hallucinated voxels as such.
 - Clean up visualization a bit in ESDF server.
 - Set default minimum ESDF distance to half truncation distance, which should lead to more accurate results.
 - **Most important**: fix bug with clear spheres which was causing erroneous lines on the border of previously-seen voxels when clearing space.

### CODE REVIEW

**Comment by helenol** [comment]
Okay found correct fix (different 3 lines than I had before), SDF tests now continue to pass, please review. :)

**Comment by helenol** [comment]
@ZacharyTaylor

File: voxblox_ros/include/voxblox_ros/ros_params.h [hunk 1]
```diff
@@ -119,7 +119,7 @@ inline EsdfIntegrator::Config getEsdfIntegratorConfigFromRosParam(
       getTsdfIntegratorConfigFromRosParam(nh_private);
 
   esdf_integrator_config.min_distance_m =
-      tsdf_integrator_config.default_truncation_distance;
+      tsdf_integrator_config.default_truncation_distance / 2.0;
```
Comment (ZacharyTaylor) [inline voxblox_ros/include/voxblox_ros/ros_params.h]: is this needed or just a better default?

Comment (helenol) [reply voxblox_ros/include/voxblox_ros/ros_params.h]: Better default in my opinion (and in the paper's quantitative evaluation).

File: voxblox/src/simulation/simulation_world.cc [hunk 2]
```diff
@@ -69,8 +69,11 @@ void SimulationWorld::getPointcloudFromViewpoint(
   // Calculate transformation between nominal camera view direction and our
   // view direction. Nominal view is positive x direction.
   const Point nominal_view_direction(1.0, 0.0, 0.0);
-  const Rotation ray_rotation(Eigen::Quaternion<FloatingPoint>::FromTwoVectors(
-      nominal_view_direction, view_direction));
+  Eigen::Quaternion<FloatingPoint> rotation_quaternion =
+      Eigen::Quaternion<FloatingPoint>::FromTwoVectors(nominal_view_direction,
+                                                       view_direction);
+  rotation_quaternion.normalize();
```
[the remark this answers is not in the document]
Comment (helenol) [reply voxblox/src/simulation/simulation_world.cc]: Apparently not, or not enough for minkindr's super duper strict normalization checks. This fixed a consistent (but absurdly rare) crash I had in the evaluation benchmarks in mav_voxblox_planning.

**Pull request merged**